NitroxClient/GameLogic/Helper/VehicleChildEntityHelper.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Helper;
+
+public class VehicleChildEntityHelper
+{
+    private static readonly List<Type> interactiveChildTypes = new List<Type> // we must sync ids of these types when creating vehicles (mainly cyclops)
+    {
+        typeof(Openable),
+        typeof(CyclopsLocker),
+        typeof(Fabricator),
+        typeof(FireExtinguisherHolder),
+        typeof(StorageContainer),
+        typeof(SeamothStorageContainer),
+        typeof(VehicleDockingBay),
+        typeof(DockedVehicleHandTarget),
+        typeof(UpgradeConsole),
+        typeof(DockingBayDoor),
+        typeof(CyclopsDecoyLoadingTube),
+        typeof(BatterySource),
+        typeof(EnergyMixin)
+    };
+
+    public static List<Entity> ExtractChildren(GameObject vehicle)
+    {
+        NitroxId parentId = NitroxEntity.GetId(vehicle);
+        List<Entity> children = new List<Entity>();
+
+        string constructedObjectsName = vehicle.GetFullHierarchyPath();
+
+        foreach (Type type in interactiveChildTypes)
+        {
+            Component[] components = vehicle.GetComponentsInChildren(type, true);
+
+            for(int i = 0; i < components.Length; i++)
```

Comment:
Is there a reason to use a indexed for loop here and not just loop over components? It would remove the assignment line below and any potential array index issues

Reply:
This file is just a rename of an existing file `VehicleInteractiveIdentifierHelper".  I didn't touch any of the logic in this PR.  However, I have fully optimized it in a subsequent commit (waiting on this PR).